repository-commons/src/main/java/io/aiven/elasticsearch/repositories/RepositoryStorageIOProvider.java

```diff
@@ -75,22 +75,23 @@ private void createOrRestoreEncryptionKey(final String basePath,
                     // the storage without compression and encryption, and it doesn't use encryption key and buffer size
                     createStorageIOFor(repositorySettings, new CryptoIOProvider(null, 0) {
                         @Override
-                        public long compressAndEncrypt(final InputStream in,
-                                                       final OutputStream out) throws IOException {
-                            return Streams.copy(in, out);
+                        public InputStream decryptAndDecompress(final InputStream in) throws IOException {
+                            return in;
                         }
 
                         @Override
-                        public InputStream decryptAndDecompress(final InputStream in) throws IOException {
-                            return in;
+                        public long compressAndEncrypt(final InputStream in,
+                                                       final OutputStream out) throws IOException {
+                            return Streams.copy(in, out);
                         }
 
                     });
             final var repositoryMetadata = new EncryptedRepositoryMetadata(encryptionKeyProvider);
             if (encKeyRepoMetadata.exists(repositoryMetadataFilePath)) {
                 LOGGER.info("Restore encryption key for repository. Path: {}", repositoryMetadataFilePath);
-                final var in = encKeyRepoMetadata.read(repositoryMetadataFilePath);
-                encryptionKey = repositoryMetadata.deserialize(in.readAllBytes());
+                try (final var in = encKeyRepoMetadata.read(repositoryMetadataFilePath)) {
+                    encryptionKey = repositoryMetadata.deserialize(in.readAllBytes());
```

Comment:
Sorry, but is this encryptionKey used anywhere?

Reply:
`encryptionKey` we use for encrypt/decrypt data in the repo, it passes to `CryptoIOProvider`